PROTO_tests/tests/categorical_test.py

```diff
@@ -0,0 +1,228 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+
+
+class TestCategorical:
+    @classmethod
+    def setup_class(cls):
+        cls.non_unique_strs = [
+            "string",
+            "string1",
+            "non-string",
+            "non-string2",
+            "string",
+            "non-string",
+            "string3",
+            "non-string2",
+            "string",
+            "non-string",
+        ]
+        cls.non_unique_cat = ak.Categorical(ak.array(cls.non_unique_strs))
+
+    def create_basic_categorical(self, prefix: str = "string", size: int = 10) -> ak.Categorical:
+        return ak.Categorical(ak.array([f"{prefix} {i}" for i in range(size)]))
+
+    def test_basic_categorical(self):
+        prefix, size = "string", 10
+        cat = self.create_basic_categorical(prefix, size)
+
+        assert list(range(size)) == cat.codes.to_list() == cat.segments.to_list()
+        assert ([f"{prefix} {i}" for i in range(size)] + ["N/A"]) == cat.categories.to_list()
+        assert size == cat.size
+        assert "Categorical" == cat.objType
+
+        with pytest.raises(ValueError):
+            ak.Categorical(ak.arange(0, 5, 10))
+
+    def test_from_codes(self):
+        codes = ak.array([7, 5, 9, 8, 2, 1, 4, 0, 3, 6])
+        categories = ak.array([f"string {i}" for i in range(10)] + ["N/A"])
+
+        cat = ak.Categorical.from_codes(codes, categories)
+        assert codes.to_list() == cat.codes.to_list()
+        assert categories.to_list() == cat.categories.to_list()
+
+    def test_substring_search(self):
+        cat = self.create_basic_categorical()
+        # update to use regex flag once #2714 is resolved
```

Comment:
Noting that the PR for this issue has been approved by at least 1 person, please be sure to update the appropriate code based on which gets merged first.

Reply:
I am going to merge this, I will not in the other PR